### ISSUE TITLE
Question on method ssm.HMM (GLM weight-sign flipped)

### ISSUE THREAD

**Issue by Gentu-Ding** [lindermanlab/ssm#164]
Hi there!

Thank you very much for the ssm package! I have a question regarding the output of the fitted weights from the ssm.HMM method when I applied it to a human reward-learning task. The task is a 2-step task that humans learn to choose between two options to collect rewards. And the GLM I’m using is really studying how previous outcome affect people’s current choice (i.e. whether they stick to the rewarded option or not). Generally people in this task showed decent reward learning, that is they expressed the behaviors consistent with a positive outcome effect on their current choice of sticking to the rewarded option (outcome: 0 - no reward, 1 - reward). But the output of the fitted weights, which was returned by glmhmm.observations.params seemed to give a negative weight loading on the effect of “outcome” (Figure 1), which seems to be the opposite of the actual behaviors (the loading on the “outcome_transition" seems to be opposite too compared with people’s actual behavior). But later if I use the method glmhmm.observations.calculate_logits to check the model prediction of behaviors in different states, the results all make sense (Figure 2). 

Just would like check this with you to see what in the usage of the package might cause the potential problem of weight-sign flipping in the Figure 1 here. I’m attaching the codes I’ve used. Thank you! 

Figure 1:
![image](https://github.com/lindermanlab/ssm/assets/14297003/2fea2f08-7a23-439c-9269-000d9cc4ca20)

Figure 2:
![image](https://github.com/lindermanlab/ssm/assets/14297003/ceb8effa-1b68-4fa4-9f44-40d5a0eacbfd)

Code 1: Fit a GLM to have initialized weights for the GLM-HMM
![image](https://github.com/lindermanlab/ssm/assets/14297003/6c1762f2-76e0-4005-a1f2-d2a8060834d7)

Code 2: Fit the GLM-HMM with the initialized weight from the GLM
![image](https://github.com/lindermanlab/ssm/assets/14297003/781656d1-4d0f-4b0f-af7f-08512c2332b2)

Best regards,
Weilun

**Comment by slinderman** [comment]
Tagging @zashwood in case she has any thoughts on the sign flipping in the GLM-HMM.

**Comment by abisi** [comment]
Hi @Gentu-Ding, did you get to the bottom of this? I'm curious because I also get negative weights for a feature that I'd expect to be positively associated with the outcome.
